### PR TITLE
Multitasking view: Don't use smooth scroll events to handle mouse wheel

### PR DESCRIPTION
**Pull request by JoseExposito**
Instead of relaying on a magic value to know detect mouse wheel "clicks" use the discrete events sent by Mutter.

This will avoid breaking the multitasking view scroll once `xf86-input-libinput` 1.2.0 lands in Ubuntu with high-resolution scroll wheel support.

### CODE REVIEW

File: src/Widgets/MultitaskingView.vala [hunk 1]
```diff
@@ -178,32 +178,34 @@ namespace Gala {
...
 
-                if (active_workspace != new_workspace) {
-                    new_workspace.activate (display.get_current_time ());
-                }
+            if (active_workspace != new_workspace) {
+                new_workspace.activate (display.get_current_time ());
             }
 
-            return false;
+            return true;
```
Comment (cassidyjames) [inline src/Widgets/MultitaskingView.vala:208]: As long as we're changing this line, can we use the Gdk macro so the intent is more clear?

```suggestion
            return EVENT_STOP;
```

Comment (davidmhewitt) [reply src/Widgets/MultitaskingView.vala:208]: Eh, since this isn't a GDK event handler, I'd be tempted not to. Yes, it's almost an impossibility that will ever be anything other than `true` but since the two things aren't related, it doesn't seem right to me.

We also need to be using less GTK/GDK in Gala, because it causes issues with Wayland. Again, yes, this is just a constant so it wouldn't be part of the problem. But if we can remove GTK altogether, we can drop the dependency.

Comment (Exalm) [reply src/Widgets/MultitaskingView.vala:208]: There's `CLUTTER_EVENT_STOP`, though doesn't seem like it's in the vapi.

Comment (JoseExposito) [reply src/Widgets/MultitaskingView.vala:208]: @Exalm is right, GNOME Shell uses `Clutter.EVENT_PROPAGATE/EVENT_STOP`. I tried to do the same, but as he mentioned, they are not in the vapi.

I guess it's something we can address in a different PR.

File: src/Widgets/MultitaskingView.vala [hunk 2]
```diff
@@ -178,32 +178,34 @@ namespace Gala {
                 return true;
             }
 
-            if (scroll_event.direction != ScrollDirection.SMOOTH) {
+            if (scroll_event.direction == ScrollDirection.SMOOTH ||
```
Comment (davidmhewitt) [inline src/Widgets/MultitaskingView.vala:181]: Was this right before?

This has changed from `!= SMOOTH` to `== SMOOTH`.

So if it is a smooth scroll event, we return here and let `GestureTracker` handle it, right? But does that mean this was wrong before? Or is there something I don't understand.

Comment (JoseExposito) [reply src/Widgets/MultitaskingView.vala:181]: The key here is that, no matter what kind of device produced the scroll, Mutter sends both a smooth and a non-smooth scroll events.

For example, 1 low-resolution ("normal" mouse) mouse wheel click will generate 1 scroll event with direction UP/DOWN *and* 1 scroll event with direction SMOOTH.

A touchpad or a high-resolution scroll mouse, scrolling a bit will generate X number of SMOOTH scroll events and, after a certain threshold is reached, 1 scroll event with direction != SMOOTH.

This design is very clever, because it allows to listen to discrete scroll events without worrying about the underlying hardware.

TL;DR

Answering your question, yes, before this change, this code worked. When the SMOOTH scroll events weren't handled by the `GestureTracker` (ie, non-touchpad events), this code was triggered and, by using a magic number, we changed the workspace.

However, once `xf86-input-libinput` 1.2.0 lands in elementary OS, we will receive SMOOTH scroll events from mice, and this won't work. Also, it is a good idea to avoid magic numbers and rely on Mutter's discrete scroll emulation.
